ctapipe/io/metadata.py

```diff
@@ -226,3 +228,21 @@ def write_to_hdf5(metadata, h5file):
         warnings.simplefilter("ignore", NaturalNameWarning)
         for key, value in metadata.items():
             h5file.root._v_attrs[key] = value  # pylint: disable=protected-access
+
+
+def read_metadata(h5filename):
```

Comment:
Could we add the possibility of passing an already opened hdf5 file here and a node other than `root`?
This could then simplify a lot of the other metadata accesing functions in the io code.

For how to pass both a path or an already opened h5file, see:

```
    with ExitStack() as stack:


        if isinstance(h5file, (str, Path)):
            h5file = stack.enter_context(tables.open_file(h5file))
        elif isinstance(h5file, tables.file.File):
            pass
        else:
            raise ValueError(
                f"expected a string, Path, or PyTables "
                f"filehandle for argument 'h5file', got {h5file}"
            )
```

Reply:
Done.

Thank you for making me aware of `ExitStack` - awesome feature